src/cargo/util/mod.rs

```diff
@@ -110,3 +110,15 @@ pub fn indented_lines(text: &str) -> String {
         })
         .collect()
 }
+
+pub fn truncate_with_ellipsis(s: &str, max_width: usize) -> String {
+    // We should truncate at grapheme-boundary and compute character-widths,
+    // yet the dependencies on unicode-segmentation and unicode-width are
+    // not worth it.
```

Comment:
`unicode-width` is already a dependency, maybe we should open an issue asking for this function to be updated to use it.